Updates for V2 primitives (#711)
Qiskit/documentation: docs/run/primitives.mdx

The Qiskit primitives are defined by open-source primitive base-classes, from
which different providers can derive their own Sampler and Estimator implementations. Among the implementations
using Qiskit, you can find reference primitive implementations for local simulation in the `qiskit.primitives` module.
Providers like Qiskit Runtime enable access to appropriate systems through native implementations of
their own primitives.

> <Admonition type="note" title="Important:">

Reviewer: ```suggestion
<Admonition type="note" title="Important">
```

way to access your hardware features. It is therefore easier for your users to benefit from your hardware's
best capabilities.

## V2 primitives

Version 2 is first the major interface change since the introduction of Qiskit Runtime primitives. Based on user feedback, the new version introduces the following major changes:

* The **new interface** lets you specify a single circuit and multiple observables (if using Estimator) and parameter value sets for that circuit. 
* You can **turn on or off individual error mitigation and suppression methods**.
* You can choose to **get counts or per-shot measurements** instead of quasi-probabilities from V2 Sampler.
* Due to scaling issues, V2 Sampler does not support specifying a resilience level, and V2 Estimator supports only levels 0 - 2. 

> * To reduce the total execution time, v2 primitives only support ISA circuits. If you need more advanced transpilation, you can use the [Qiskit transpiler](advanced-runtime-options#runtime-compilation) (manually) or the [AI-driven transpilation service (premium users)](../transpile/qiskit-transpiler-service).

Reviewer 2: Slight change in wording as the original sounded like transpilation was optional. 

```suggestion
* To reduce the total execution time, v2 primitives only support ISA circuits. You can use the [Qiskit transpiler](advanced-runtime-options#runtime-compilation) (manually) or the [AI-driven transpilation service (premium users)](../transpile/qiskit-transpiler-service) to transform the circuits before making the queries to the primitives.
```

> * To reduce the total execution time, v2 primitives only support **lightweight transpilation**. If you need more advanced transpilation, you can use the [Qiskit transpiler](advanced-runtime-options#runtime-compilation) (manually) or the [AI-driven transpilation service (premium users)](../transpile/qiskit-transpiler-service).

Reviewer 2: This should say ISA circuits and not lightweight transpilation.

```

#### Estimator V2

* Takes one parameter: PUBs in the format (`<circuit>`, `<observables>`, `<parameter values>`, `<precision>`)
* Numpy [broadcasting rules](https://numpy.org/doc/stable/user/basics.broadcasting.html) are used when combining observables and parameter values.
* Each input PUB has a corresponding PubResult that contains both data and metadata.

Example: 

```python
estimator.run([(circuit1, observable1, param_values1),(circuit2, observable2, param_values2) ])
```

#### Sampler V1

* Takes two parameters: circuits and parameter values.
* Each parameter can be a single value or a list.
* The length of the parameters must match.
* Elements from each are aggregated.

Example: 

```python
sampler.run([circuit1, circuit2, ...],[observable1, observable2, ...],[param_values1, param_values2, ...] )
```

#### Sampler V2

* Takes one parameter: PUBs in the format  (`<circuit>`, `<parameter values>`, `<shots>`)
* Elements from each are aggregated.
* Obeys program outputs. Typically this is a bit array but can also be an array of complex numbers (measurement level 1). 

> * Returns raw data type. Data from each shot is returned (memory=True), and post-processing is done by using convenience methods.

Reviewer 2: ```suggestion
* Returns raw data type. Data from each shot is returned (analogous to `memory=True` in the `backend.run` interface), and post-processing is done by using convenience methods.
```

* Output data is grouped by output registers.

Example: 

```python
sampler.run([(circuit1, param_values1, shots1),(circuit2, param_values2, shots_2) ])
```

## Estimator

The Estimator behaves differently, depending on what version you are using. 

<Tabs>
  <TabItem value="EstimatorV2" label="EstimatorV2">
    EstimatorV2 computes expectation values of observables with respect to states prepared by quantum circuits.
It receives one or more PUBs as the inputs and returns the computed expectation values per pair, along with their

> variances, in `PubResult` form. Different Estimator implementations support various configuration options. The circuits

Reviewer 2: ```suggestion
standard error, in `PubResult` form. Different Estimator implementations support various configuration options. The circuits
```

can be parametrized, as long as the parameter values are also provided as input to the primitive.
  </TabItem>

  <TabItem value="EstimatorV1" label="Estimator (V1)">
    The Estimator primitive computes expectation values of observables with respect to states prepared by quantum circuits.
The Estimator receives circuit-observable pairs (with the observable expressed as a
weighted sum of Pauli operators) as inputs, and returns the computed expectation values per pair, as well as their
variances. Different Estimator implementations support various configuration options. The circuits
can be parametrized, as long as the parameter values are also provided as input to the primitive.
  </TabItem>
</Tabs>

## Sampler

The Sampler behaves differently, depending on what version you are using. 

<Tabs>
  <TabItem value="SamplerV2" label="SamplerV2">

>     SamplerV2 is simplified to focus on its core task of sampling the output register from execution of quantum circuits. It returns the samples, whose type is defined by the program, without weights and therefore does not support resilience levels. The result class, however, has methods to return weighted samples, such as counts and quasi-probabilities. 

Reviewer 2: ```suggestion
    SamplerV2 is simplified to focus on its core task of sampling the output register from execution of quantum circuits. It returns the samples, whose type is defined by the program, without weights and therefore does not support resilience levels. The result class, however, has methods to return weighted samples, such as counts.
```

    
    It receives one or more PUBs as the inputs and returns quasi-probabilities, and optionally counts or per-shot measurements, as `PubResult`s.  Quasi-probability distributions are similar to regular
probabilities, except they might include negative values, which can occur when using certain error mitigation techniques. The circuits

> can be parametrized, as long as the parameter values are also provided as input to the primitive.

Reviewer 2: V2 doesn't return quasi-probabilities

way to access your hardware features. It is therefore easier for your users to benefit from your hardware's
best capabilities.

## V2 primitives

Version 2 is first the major interface change since the introduction of Qiskit Runtime primitives. Based on user feedback, the new version introduces the following major new functions:

> * A **new interface** that allows you to specify a single circuit and multiple observables (if using Estimator) and parameter value sets for that circuit. 

Reviewer: ```suggestion
* A **new interface** that lets you specify a single circuit and multiple observables (if using Estimator) and parameter value sets for that circuit. 
```

#### Estimator V1

* Takes three parameters: circuits, observables, and parameter values.
* Each parameter can be a single value or a list.
* The length of the parameters must match.
* Elements from each are aggregated.

Example: 

```python
estimator.run([circuit1, circuit2, ...],[observable1, observable2, ...],[param_values1, param_values2, ...] )
```

#### Estimator V2

* Takes one parameter: PUBs in the format (`<circuit>`, `<observables>`, `<parameter values>`, `<precision>`)
* Numpy [broadcasting rules](https://numpy.org/doc/stable/user/basics.broadcasting.html) are used when combining observables and parameter values.
* Each input PUB has a corresponding PubResult that contains both data and metadata.

Example: 

```python
estimator.run([(circuit1, observable1, param_values1),(circuit2, observable2, param_values2) ])
```

#### Sampler V1

> 

Reviewer: flagging for missing content under the header here

> 1. `Sampler` and `Estimator`

Reviewer 2: V2 reference implementation are called `StatevectorSampler` and `StatevectorEstimator`
